app/services/filter_manager.rb

```diff
@@ -0,0 +1,78 @@
+class FilterManager
+  attr_reader :current_user, :session, :params, :filter_type
+
+  def initialize(current_user:, session:, params:, filter_type:)
+    @current_user = current_user
+    @session = session
+    @params = params
+    @filter_type = filter_type
+  end
+
+  def self.filter_by_search(base_collection, search_term = nil)
+    if search_term.present?
+      base_collection.search_by(search_term)
+    else
+      base_collection
+    end
+  end
+
+  def self.filter_logs(logs, search_term, filters, all_orgs, user)
+    logs = filter_by_search(logs, search_term)
+
+    filters.each do |category, values|
+      next if Array(values).reject(&:empty?).blank?
```

Comment:
Array(values).compact_blank?
I think

Reply:
good point this can be simplified, have gone for values.all(&:empty?)